packages/remark-lint-no-undefined-references/index.js

```diff
@@ -35,7 +35,7 @@ module.exports = rule(
 var reason = 'Found reference to undefined definition'
 
 function noUndefinedReferences(tree, file) {
-  var map = {}
+  var map = Object.create(null)
```

Comment:
This change, in conjunction with the `in` check, prevents the possibility of false positives caused by inherited properties. It is unrelated to the primary change, so I will be happy to move it to a separate pull request if you would like me to do so.

Reply:
> in this case it doesn’t matter because we [`toUpperCase`][1] keys

The problem with code whose semantics do not match the author's intent is that readers must sometimes infer intent from the code's semantics. In this case it was not clear to me whether `String#toUpperCase` is being used to avoid collisions with names of inherited properties or because link identifiers are case-insensitive.


[1]: https://github.com/remarkjs/remark-lint/blob/3dc4fb0f43e74a7ae86996baeea581b284ce2077/packages/remark-lint-no-undefined-references/index.js#L53

Reply:
Oh totez! This is several years old, when hacks like those were necessary. But I’m not saying it’s good and I agree with you!